test/Conversion/FIRRTLToRTL/lower-to-rtl.mlir

```diff
@@ -615,55 +615,54 @@ module attributes {firrtl.mainModule = "Simple"} {
+    // CHECK-NEXT:   sv.connect %_M_read_data, %[[data]] : i42
     // CHECK-NEXT: } else  {
-    // CHECK-NEXT:   %2 = sv.read_inout %_M_read_addr : !rtl.inout<i4>
-    // CHECK-NEXT:   %3 = sv.array_index_inout %_M[%2]
-    // CHECK-NEXT:   %4 = sv.read_inout %3 : !rtl.inout<i42>
     // CHECK-NEXT:   %c-4_i4 = rtl.constant(-4 : i4) : i4
-    // CHECK-NEXT:   %5 = rtl.icmp ult %2, %c-4_i4 : i4
-    // CHECK-NEXT:   %6 = sv.textual_value "`RANDOM" : i42
-    // CHECK-NEXT:   %7 = rtl.mux %5, %4, %6 : i42
-    // CHECK-NEXT:   sv.connect %_M_read_data, %7 : i42
+    // CHECK-NEXT:   %[[cond:.+]] = rtl.icmp ult %[[addr]], %c-4_i4 : i4
+    // CHECK-NEXT:   %[[random:.+]] = sv.textual_value "`RANDOM" : i42
+    // CHECK-NEXT:   %[[dataOrRandom:.+]] = rtl.mux %[[cond]], %[[data]], %[[random]] : i42
+    // CHECK-NEXT:   sv.connect %_M_read_data, %[[dataOrRandom]] : i42
+    // CHECK-NEXT: }
+
+    // COM: Write port.
+    // COM: --------------------------------------------------------------------
+    // CHECK-DAG: %_M_write_addr = sv.wire : !rtl.inout<i4>
+    // CHECK-DAG: %_M_write_en = sv.wire : !rtl.inout<i1>
+    // CHECK-DAG: %_M_write_clk = sv.wire : !rtl.inout<i1>
+    // CHECK-DAG: %_M_write_data = sv.wire : !rtl.inout<i42>
+    // CHECK-DAG: %_M_write_mask = sv.wire : !rtl.inout<i1>
+
+    // CHECK:      %[[clk:.+]] = sv.read_inout %_M_write_clk
+    // CHECK:      sv.alwaysff(posedge %[[clk]]) {
+    // CHECK-DAG:    %[[en:.+]] = sv.read_inout %_M_write_en : !rtl.inout<i1>
+    // CHECK-DAG:    %[[mask:.+]] = sv.read_inout %_M_write_mask : !rtl.inout<i1>
+    // CHECK:        %[[cond:.+]] = rtl.and %[[en]], %[[mask]] : i1
+    // CHECK-NEXT:   sv.if %[[cond]]  {
+    // CHECK:          %[[addr:.+]] = sv.read_inout %_M_write_addr : !rtl.inout<i4>
+    // CHECK-DAG:      %[[mem:.+]] = sv.array_index_inout %_M[%[[addr]]] : !rtl.inout<uarray<12xi42>>, i4
+    // CHECK-DAG:      %[[data:.+]] = sv.read_inout %_M_write_data
+    // CHECK:          sv.passign %[[mem]], %[[data]] : i42
+    // CHECK-NEXT:   }
     // CHECK-NEXT: }
 
-    // Write port.
-    // CHECK:      %_M_write_addr = sv.wire : !rtl.inout<i4>
-    // CHECK-NEXT: %_M_write_en = sv.wire : !rtl.inout<i1>
-    // CHECK-NEXT: %_M_write_clk = sv.wire : !rtl.inout<i1>
-    // CHECK-NEXT: %_M_write_data = sv.wire : !rtl.inout<i42>
-    // CHECK-NEXT: %_M_write_mask = sv.wire : !rtl.inout<i1>
-
-    // CHECK-NEXT: %0 = sv.read_inout %_M_write_clk
-    // CHECK-NEXT: sv.always posedge %0  {
-    // CHECK-NEXT:     %2 = sv.read_inout %_M_write_en : !rtl.inout<i1>
-    // CHECK-NEXT:     %3 = sv.read_inout %_M_write_mask : !rtl.inout<i1>
-    // CHECK-NEXT:     %4 = rtl.and %2, %3 : i1
-    // CHECK-NEXT:     sv.if %4  {
-    // CHECK-NEXT:       %5 = sv.read_inout %_M_write_data : !rtl.inout<i42>
-    // CHECK-NEXT:       %6 = sv.read_inout %_M_write_addr : !rtl.inout<i4>
-    // CHECK-NEXT:       %7 = sv.array_index_inout %_M[%6]
-    // CHECK-NEXT:       sv.bpassign %7, %5 : i42
```

Comment:
I believe switching blocking assign to non-blocking addresses the first warning here: https://github.com/llvm/circt/issues/543#issuecomment-777964502.

Reply:
Yeah, you're right. I noticed that it was weird that blocking assigns were being used here. 

Fixed that just now on `main` with 4940394.